schema/ksl-schema.kdl

```diff
@@ -0,0 +1,925 @@
+                                    schema.
+
+                                    Relative schema references SHALL be resolved
+                                    against `metadata > id`.
+                                    """ {
+                                    type string
+                                    format url-reference irl-reference
+                                }
+                                children {
+                                    node id about="ID reference to another node." {
+                                        repeatable
+                                        arg {
+                                            type string
+                                        }
+
+                                    }
+                                    node path about="KPath reference to another node." {
+                                        repeatable
+                                        arg {
+                                            type string
+                                            format kpath
+                                        }
+                                    }
+                                }
+                            }
+                            // TODO: be more precise and clear about merge + override behavior.
+                            node default about="""
+                                Definition default
+
+                                Any children of this node will be merged into the node
+                                definition itself, as if written directly into the node.
+                                Any conflicting definitions will be overriden by
+                                definitions within the rest of the node.
+                                """ {
+                                children {
+                                    allow-others
+                                }
+                            }
+                            node undefine about="Undefine a node with this name" {
+                                arg {
+                                    optional
+                                    type boolean
+                                    default #true
+                                }
+                            }
+                            node required about="""
+                                Node is required
+
+                                By default, all declared child nodes are
+                                optional. Including this option will require
+                                that this node always appear in its parent's
+                                children block.
+                                """ {
+                                arg {
+                                    type boolean
+                                    default #true
+                                }
+                            }
+                            node repeatable about="""
```

Comment:
I don't know why I have mixed feelings about this name, and I don't know if I can think of anything I like more, but in case you have an opinion, the closest I can get to happy is `allow-multiple`?

Reply:
I workshopped a few names (like `multiple`, and `repeatable` ended up feeling the best to me when I was actually writing out a schema